storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -8599,23 +8599,37 @@ handle_options(int argc, char **argv, int *argc_client, char ***argv_client,
 	for (int i = 0 ; i < *argc_client ; i++) {
 		const char * const opt = (*argv_client)[i];
 
-		if (strncmp(opt, "--", 2) &&
-		    !(strlen(opt) == 2 && opt[0] == '-')) {
-			bool server_option = true;
+		bool server_option = true;
 
-			for (int j = 0; j < *argc_server; j++) {
-				if (opt == (*argv_server)[j]) {
-					server_option = false;
-					break;
-				}
+		for (int j = 0; j < *argc_server; j++) {
+			if (opt == (*argv_server)[j]) {
+				server_option = false;
+				break;
 			}
+		}
+
+		if (strncmp(opt, "--", 2) &&
+		    !(strlen(opt) == 2 && opt[0] == '-')) {
 
 			if (!server_option) {
 				msg("xtrabackup: Error:"
 				    " unknown argument: '%s'\n", opt);
 				exit(EXIT_FAILURE);
 			}
 		}
+
+			bool client_option = true;
+			for (unsigned j = 0; j < xb_client_options_count; j++) {
+				if (strcmp(opt, (xb_client_options)[j].name)) {
```

Comment:
Same questions, I would like to see example output and the test case. Also, can we abort with error if incorrect argument has been provided via command line?

Reply:
For the error message: the ticket asked for a warning, to avoid issues if we do not recognize a valid command line option. I could change it to an error if that's better, but in that case, we probably need a more detailed testing for it to avoid any issues.

For the test case: I'll rebase this on #495 after that's approved, to use the same testcase.